src/lsm/tree.zig

```diff
@@ -701,28 +787,76 @@ pub fn TreeType(comptime Table: type, comptime Storage: type, comptime tree_name
             assert(tree.compaction_callback != null);
 
             // compact_done() is called after all compact_tick()'s complete.
-            // This function can be triggered asynchronously or by compact_cpu() below.
             tree.compaction_io_pending -= 1;
             if (tree.compaction_io_pending == 0) tree.compact_done();
         }
 
+        /// Called at the end of each compaction tick.
         fn compact_done(tree: *Tree) void {
```

Comment:
@kprotty Would you please review everything here in `compact_done()` since you are most familiar?